src/openrct2/world/Park.cpp

```diff
@@ -72,1007 +82,1089 @@ sint32 _suggestedGuestMaximum;
-        return;
+    for (size_t i = 0; i < 20; i++)
+    {
+        gMarketingCampaignDaysLeft[i] = 0;
     }
 
-    format_string(oldName, 128, gParkName, &gParkNameArgs);
-    if (strcmp(oldName, newName) == 0) {
-        *ebx = 0;
-        return;
-    }
+    research_reset_items();
+    finance_init();
 
-    if (newName[0] == 0) {
-        gGameCommandErrorText = STR_INVALID_RIDE_ATTRACTION_NAME;
-        *ebx = MONEY32_UNDEFINED;
-        return;
+    for (size_t i = 0; i < 8; i++)
+    {
+        gResearchedRideTypes[i] = 0;
     }
 
-    newUserStringId = user_string_allocate(USER_STRING_HIGH_ID_NUMBER, newName);
-    if (newUserStringId == 0) {
-        gGameCommandErrorText = STR_INVALID_NAME_FOR_PARK;
-        *ebx = MONEY32_UNDEFINED;
-        return;
-    }
+    reset_researched_scenery_items();
 
-    if (*ebx & GAME_COMMAND_FLAG_APPLY) {
-        // Log park rename command if we are in multiplayer and logging is enabled
-        if ((network_get_mode() == NETWORK_MODE_CLIENT || network_get_mode() == NETWORK_MODE_SERVER) && gConfigNetwork.log_server_actions) {
-            // Get player name
-            int player_index = network_get_player_index(game_command_playerid);
-            const char* player_name = network_get_player_name(player_index);
+    gParkEntranceFee = MONEY(10, 00);
+    gPeepSpawns[0].x = PEEP_SPAWN_UNDEFINED;
+    gPeepSpawns[1].x = PEEP_SPAWN_UNDEFINED;
+    gResearchPriorities =
+        (1 << RESEARCH_CATEGORY_TRANSPORT) |
+        (1 << RESEARCH_CATEGORY_GENTLE) |
+        (1 << RESEARCH_CATEGORY_ROLLERCOASTER) |
+        (1 << RESEARCH_CATEGORY_THRILL) |
+        (1 << RESEARCH_CATEGORY_WATER) |
+        (1 << RESEARCH_CATEGORY_SHOP) |
+        (1 << RESEARCH_CATEGORY_SCENERYSET);
+    gResearchFundingLevel = RESEARCH_FUNDING_NORMAL;
 
-            char log_msg[256];
-            char* args[3] = {
-                (char *) player_name,
-                oldName,
-                newName
-            };
-            format_string(log_msg, 256, STR_LOG_PARK_NAME, args);
-            network_append_server_log(log_msg);
-        }
+    gGuestInitialCash = MONEY(50,00);
+    gGuestInitialHappiness = CalculateGuestInitialHappiness(50);
+    gGuestInitialHunger = 200;
+    gGuestInitialThirst = 200;
+    gScenarioObjectiveType = OBJECTIVE_GUESTS_BY;
+    gScenarioObjectiveYear = 4;
+    gScenarioObjectiveNumGuests = 1000;
+    gLandPrice = MONEY(90,00);
+    gConstructionRightsPrice = MONEY(40,00);
+    gParkFlags = PARK_FLAGS_NO_MONEY | PARK_FLAGS_SHOW_REAL_GUEST_NAMES;
+    ResetHistories();
+    finance_reset_history();
+    award_reset();
 
-        // Free the old ride name
-        user_string_free(gParkName);
-        gParkName = newUserStringId;
+    gS6Info.name[0] = '\0';
+    format_string(gS6Info.details, 256, STR_NO_DETAILS_YET, nullptr);
+}
 
-        gfx_invalidate_screen();
-    } else {
-        user_string_free(newUserStringId);
+void Park::Update(const Date &date)
+{
+    // Every ~5 seconds
```

Comment:
Out of curiosity, why did you review your own pr instead of making a fix commit?  Or was this an edit from someone else? Or are you just leaving a note to fix later? Does it benefit teamwise any when you do a comment review on your pr?